src/topology/builder.rs

```diff
@@ -272,7 +272,9 @@ pub async fn build_pieces(
 
         let task = Task::new(id.clone(), typetag, sink);
 
-        let component_id = id.to_string();
+        let component_name = id.to_string();
+        let component_id = id.id().to_string();
```

Comment:
How do these two differ? Could you given an example of both?

Reply:
`id.to_string()` would give `pipeline_id#transform_name` but `id.id().to_string()` would just give `transform_name`.
That's why `component_scope` is added after to have the complete information.

Reply:
Ah, gotcha, I see. `component_name` is currently a deprecated tag that is always meant to match `component_id` to let users migrate over easily (as part of #8417). Given that, I think we should keep them matching here.

Reply:
I'm not sure I'd agree with that but I might be wrong. I'd say`component_name` is the behavior of having a single field whereas `component_id` would be the combination with `component_scope`. Therefore, we should have all the information in `component_name` that we have with `component_id` and `component_scope` combined 🤔

Reply:
🤔 in #8417 we decided to just use `id` and deprecate `name`. `name` was left as a span field purely for backwards compatibility. One idea was that in the future users could perhaps specify human readable names as a `name = "Some Name"` config on components. We could decide to revive `name` with your definition, but I'm wondering if we could come up with a better term. Maybe `qualified_id`?

Reply:
🤔 Ok. Maybe not to block this PR even more, I'll change what you asked for and we can see that afterward